ahnlich/types/src/query.rs

```diff
@@ -69,6 +76,25 @@ pub enum Query {
     ListClients,
     Ping,
 }
+fn serialize_non_zero_usize<S>(value: &NonZeroUsize, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    serializer.serialize_u64(value.get() as u64)
+}
+
+fn deserialize_non_zero_usize<'de, D>(deserializer: D) -> Result<NonZeroUsize, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let mut value = u64::deserialize(deserializer)?;
+    if value < 1 {
+        value += 1
```

Comment:
We shouldn't sneakily add stuff for them
I'd rather the deserialization fails than do something sneaky like turning 0s to 1s

Reply:
We might later catch deserialization errors to report back to the client (although we don't currently do that, we just break)

Reply:
The issue here is, tracer adds a default value of 0 abandoning the default value specified causing nonzerousize to break

Reply:
Let me try creating a sample, just to confirm

Reply:
Yeah, creating samples resolves this, so no need for extra serialization and deserialization functions